diff --git src/sysc/packages/boost/bind/placeholders.hpp src/sysc/packages/boost/bind/placeholders.hpp
index 20c5e61..3f3ca37 100644
--- src/sysc/packages/boost/bind/placeholders.hpp
+++ src/sysc/packages/boost/bind/placeholders.hpp
@@ -25,7 +25,7 @@
 namespace
 {
 
-#if defined(__BORLANDC__)
+#if defined(__BORLANDC__) || defined(__GNUC__)
 
 static inline boost::arg<1> _1() { return boost::arg<1>(); }
 static inline boost::arg<2> _2() { return boost::arg<2>(); }
