diff --git src/sysc/datatypes/bit/sc_bit_proxies.h src/sysc/datatypes/bit/sc_bit_proxies.h
index 40bff6f..974302f 100644
--- src/sysc/datatypes/bit/sc_bit_proxies.h
+++ src/sysc/datatypes/bit/sc_bit_proxies.h
@@ -713,7 +713,7 @@ public:
 
 protected:
 
-    mutable X& m_obj;
+    X& m_obj;
     int        m_hi;
     int        m_lo;
     int        m_len;
@@ -1190,10 +1190,10 @@ public:
 
 protected:
 
-    mutable X&   m_left;
-    mutable Y&   m_right;
+    X&   m_left;
+    Y&   m_right;
     mutable int  m_delete;
-    mutable int& m_refs;
+    int& m_refs;
 
 private:
 
